src/core/ext/transport/chttp2/server/chttp2_server.cc

```diff
@@ -373,15 +373,20 @@ void Chttp2ServerListener::ActiveConnection::HandshakingState::OnHandshakeDone(
   HandshakingState* self = static_cast<HandshakingState*>(args->user_data);
   OrphanablePtr<HandshakingState> handshaking_state_ref;
   RefCountedPtr<HandshakeManager> handshake_mgr;
-  bool cleanup_connection = false;
+  bool remove_connection = false;
+  OrphanablePtr<ActiveConnection> connection;
   grpc_resource_user* resource_user =
       self->connection_->listener_->server_->default_resource_user();
   {
     MutexLock connection_lock(&self->connection_->mu_);
     if (error != GRPC_ERROR_NONE || self->connection_->shutdown_) {
       const char* error_str = grpc_error_string(error);
       gpr_log(GPR_DEBUG, "Handshaking failed: %s", error_str);
-      cleanup_connection = true;
+      remove_connection = true;
+      if (resource_user != nullptr) {
+        grpc_resource_user_free(resource_user,
```

Comment:
Three instances of grpc_resource_user_free in this code replaced what was previously one instance. That seems like a step backwards. Is it unavoidable here?